src/contexts/providers.context.tsx

```diff
@@ -184,54 +181,47 @@ const ProvidersProvider: FC = (props) => {
   );
 
   const changeNetwork = useCallback(
-    async (chain: Chain) => {
+    (chain: Chain) => {
       if (
         env &&
         connectedProvider &&
-        connectedProvider.provider.isMetaMask &&
-        connectedProvider.provider.request
+        connectedProvider.provider.provider.isMetaMask &&
+        connectedProvider.provider.provider.request
       ) {
-        const request = connectedProvider.provider.request;
-        try {
-          const network = await chain.provider.getNetwork();
-          await request({
+        const request = connectedProvider.provider.provider.request;
+        return (
+          request({
             method: "wallet_switchEthereumChain",
-            params: [{ chainId: hexValue(network.chainId) }],
-          });
-        } catch (error) {
-          try {
-            const network = await chain.provider.getNetwork();
-            if (isMetamaskUnknownChainError(error)) {
-              await request({
-                method: "wallet_addEthereumChain",
-                params: [
-                  {
-                    chainId: hexValue(network.chainId),
-                    chainName: getChainName(chain),
-                    rpcUrls: [chain.provider.connection.url],
-                  },
-                ],
-              });
-            } else {
-              throw error;
-            }
-          } catch (error) {
-            if (isMetamaskUserRejectedRequestError(error)) {
-              throw new Error("User rejected the request.");
-            }
-            throw new Error("Unknown error from Metamask");
-          }
-        }
-      } else {
-        return Promise.reject(new Error(providersContextNotReadyErrorMsg));
+            params: [{ chainId: hexValue(chain.chainId) }],
+          })
+            // eslint-disable-next-line @typescript-eslint/no-empty-function
+            .then(() => {})
```

Comment:
Just do: `.then(() => ({}))` so we can delete the `eslint-disable-next-line` above.

Reply:
If we add the parenthesis we'll return and object instead of a function